From b9aa8b79270bb5241508b2c3864c8069f73e75e5 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Thu, 20 Nov 2025 16:39:06 +0100
Subject: [PATCH 1/2] Permit to provide custom CPPFLAGS_FOR_BUILD

In the context of cross compilation the CPPFLAGS for the target might be
different than the one for build tools.

Permit to provide custom CPPFLAGS_FOR_BUILD value to handle this
scenario.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/configure.ac
+++ b/configure.ac
@@ -61,7 +61,7 @@ then
     # are not the same, we set reasonable default values for the tools.
 
     CC_FOR_BUILD=${CC_FOR_BUILD-gcc}
-    CPPFLAGS_FOR_BUILD="\$(CPPFLAGS)"
+    CPPFLAGS_FOR_BUILD=${CPPFLAGS_FOR_BUILD-"\$(CPPFLAGS)"}
     CC=${CC-${host_alias}-gcc}
     CFLAGS=${CFLAGS-"-g -O2"}
     CXX=${CXX-${host_alias}-c++}
